Use '.Values.enableIPv4Masquerade' in 'node-init'

bmcustodio opened the pull request:
I'm not 100% sure if this was overlooked during the
`masquerade` → `enableIPv4Masquerade` renaming
(https://github.com/cilium/cilium/pull/14124), but assuming that was
the case, this commit changes the reference from the old value to the
new one so the value of `enableIPv4Masquerade` is actually taken into
account in `node-init`.

```release-note
`node-init` now takes `enableIPv4Masquerade` into account on GKE.
```

sayboras commented on install/kubernetes/cilium/files/nodeinit/prestop.bash:
```diff
@@ -46,7 +46,7 @@ echo "Restarting the kubelet"
 {{- end }}
 
-{{- if (and .Values.gke.enabled (or .Values.masquerade .Values.gke.disableDefaultSnat))}}
+{{- if (and .Values.gke.enabled (or .Values.enableIPv4Masquerade .Values.gke.disableDefaultSnat))}}
```
while you are here, do you know where .Values.gke.disableDefaultSnat comes from ?

bmcustodio replied:
I had to go a bit back in time to find it  — [`v1.8`](https://github.com/cilium/cilium/blob/v1.8/install/kubernetes/cilium/values.yaml#L448), it seems. It's absent in `v1.9` and later. I wonder if it can be removed too. 🤔

EronWright replied:
For posterity, the `gke.disableDefaultSnat` value was introduced here: https://github.com/cilium/cilium/pull/12952

When true, it instructs nodeinit to remove the default IP-MASQ chain rules that GKE installs into the node.  It is redundant if `enableIPv4Masquerade` is true.

sayboras reviewed (approved):
LGTM :100:

bmcustodio committed:
Use '.Values.enableIPv4Masquerade' in 'node-init'.

I'm not 100% sure if this was overlooked during the
`masquerade` → `enableIPv4Masquerade` renaming
(https://github.com/cilium/cilium/pull/14124), but assuming that was
the case, this commit changes the reference from the old value to the
new one so the value of `enableIPv4Masquerade` is actually taken into
account in `node-init`.

Signed-off-by: Bruno M. Custódio <brunomcustodio@gmail.com>
